api/editions_test.go

```diff
@@ -43,6 +44,150 @@ func TestGetEditionsReturnsOK(t *testing.T) {
+		w := httptest.NewRecorder()
+		mockedDataStore := &storetest.StorerMock{
+			CheckDatasetExistsFunc: func(datasetID, state string) error {
+				return nil
+			},
+			GetEditionsFunc: func(ctx context.Context, id string, state string, offset, limit int, authorised bool) (*models.EditionUpdateResults, error) {
+				return &models.EditionUpdateResults{
+					Items: []*models.EditionUpdate{
+						{ID: "id1",
+							Current: &models.Edition{
+								ID: "id2",
+							}},
+					},
+					Count:      1,
+					Offset:     offset,
+					Limit:      limit,
+					TotalCount: 3,
+				}, nil
+			},
+		}
+
+		datasetPermissions := getAuthorisationHandlerMock()
+		permissions := getAuthorisationHandlerMock()
+		api := GetAPIWithMocks(mockedDataStore, &mocks.DownloadsGeneratorMock{}, datasetPermissions, permissions)
+		api.Router.ServeHTTP(w, r)
+
+		Convey("Then the call succeeds with 200 OK code, expected body and calls", func() {
+			expectedResponse := models.EditionUpdateResults{
+				Items: []*models.EditionUpdate{
+					{ID: "id2"},
+				},
+				Count:      1,
+				Offset:     2,
+				Limit:      2,
+				TotalCount: 3,
+			}
+
+			So(w.Code, ShouldEqual, http.StatusOK)
+			validateBody(w.Body.Bytes(), expectedResponse)
+		})
+	})
+
+	Convey("When valid limit above maximum and offset query parameters are provided, then return editions information according to the offset and limit", t, func() {
+
+		r := httptest.NewRequest("GET", "http://localhost:22000/datasets/123-456/editions?offset=2&limit=7", nil)
+		w := httptest.NewRecorder()
+		mockedDataStore := &storetest.StorerMock{
+			CheckDatasetExistsFunc: func(datasetID, state string) error {
+				return nil
+			},
+			GetEditionsFunc: func(ctx context.Context, id string, state string, offset, limit int, authorised bool) (*models.EditionUpdateResults, error) {
+				return &models.EditionUpdateResults{
+					Items: []*models.EditionUpdate{
+						{ID: "id1",
+							Current: &models.Edition{
+								ID: "id2",
+							}},
+					},
+					Count:      1,
+					Offset:     offset,
+					Limit:      limit,
+					TotalCount: 3,
+				}, nil
+			},
+		}
+
+		datasetPermissions := getAuthorisationHandlerMock()
+		permissions := getAuthorisationHandlerMock()
+		api := GetAPIWithMocks(mockedDataStore, &mocks.DownloadsGeneratorMock{}, datasetPermissions, permissions)
+		api.Router.ServeHTTP(w, r)
+
+		Convey("Then the call succeeds with 200 OK code, expected body and calls", func() {
+			expectedResponse := models.EditionUpdateResults{
+				Items: []*models.EditionUpdate{
+					{ID: "id2"},
+				},
+				Count:      1,
+				Offset:     2,
+				Limit:      7,
+				TotalCount: 3,
+			}
+
+			So(w.Code, ShouldEqual, http.StatusOK)
+			validateBody(w.Body.Bytes(), expectedResponse)
+		})
+	})
+
+	Convey("When a negative limit and offset query parameters are provided, then return editions information with offset and limit equal to zero", t, func() {
```

Comment:
I would expect negative values to return 400 bad request status with a message saying these query parameters are invalid, must be positive